src/Emulator/Peripherals/Peripherals/Timers/OpenTitan_Timer.cs

```diff
@@ -0,0 +1,201 @@
+                         {
+                            this.Log(LogLevel.Noisy, $"Get timer0 compare upper - CompareValue0: 0x{compareValue0:X}");
+                            return (uint)((compareValue0 >> 32) & (ulong)0x00000000_FFFFFFFF);
+                         })},
+
+                 {(long)Registers.InterruptEnable0, new DoubleWordRegister(this)
+                 .WithFlag(0, out timerInterruptEnable0, name: "timer0 interrupt enable", changeCallback: (_, value) =>
+                         {
+                            string enable_message = value ? "Enabled" : "Disabled";
+                            this.Log(LogLevel.Noisy, $"{enable_message} timer0 interrupt");
+                            EventEnabled = value;
+                         })
+                 .WithIgnoredBits(1, 31)},
+
+                 {(long)Registers.InterruptState0, new DoubleWordRegister(this)
+                 .WithFlag(0, FieldMode.Read | FieldMode.Write, name: "timer0 innerupt state", writeCallback: (_, value) =>
+                         {
+                            string message = value ? "Request Clear" : "Unset";
+                            this.Log(LogLevel.Noisy, $"{message} timer0 interrupt");
+                            if (value)
+                            {
+                                timerExpired0IRQ.Set(false);
+                            }
+                         }, valueProviderCallback: _ =>
+                         {
+                            return timerExpired0IRQ.IsSet;
+                         })
+                 .WithIgnoredBits(1, 31)},
+           };
+           registers = new DoubleWordRegisterCollection(this, registersMap);
+
+        }
+
+        public override void Reset()
+        {
+            base.Reset();
+            timerExpired0IRQ.Set(false);
+        }
+
+        public GPIO IRQ { get; } = new GPIO();
+
+        public long Size => 0x200;
+
+        public ulong TimerValue => Value;
+
+        public uint ReadDoubleWord(long offset)
+        {
+            return registers.Read(offset);
+        }
+
+        public void WriteDoubleWord(long offset, uint value)
+        {
+            registers.Write(offset, value);
+        }
+
+        private readonly DoubleWordRegisterCollection registers;
+
+        private enum Registers
+       {
+            Control = 0x0,
+            Config0 = 0x100,
+            TimerValueLower0 = 0x104,
+            TimerValueUpper0 = 0x108,
+            CompareLower0 = 0x10c,
+            CompareUpper0 = 0x110,
+            InterruptEnable0 = 0x114,
+            InterruptState0 = 0x118,
+            InterruptTest0 = 0x11c,
+        }
+
+        private IFlagRegisterField timerActive0;
+        private uint prescalerValue0;
+        private uint stepValue0;
+        private ulong timerValue0;
+        private ulong compareValue0;
+        private IFlagRegisterField timerInterruptEnable0;
+
+        public GPIO timerExpired0IRQ
```

Comment:
public property should be named with uppercase - `TimerExpired0IRQ`
additionally, the `public GPIO TimerExpired0IRQ { get; }` should be enough (this will make the property readonly)

Reply:
Done.